nym-vpn-x/src-tauri/src/grpc/client.rs

```diff
@@ -132,3 +170,42 @@ impl From<ServingStatus> for VpndStatus {
         }
     }
 }
+
+async fn get_channel(socket_path: PathBuf) -> anyhow::Result<Channel> {
+    // NOTE the uri here is ignored
+    Ok(TonicEndpoint::from_static(DEFAULT_HTTP_ENDPOINT)
+        .connect_with_connector(tower::service_fn(move |_| {
+            IpcEndpoint::connect(socket_path.clone())
+        }))
+        .await?)
+}
+
+impl Default for Transport {
+    fn default() -> Self {
+        Transport::Ipc(DEFAULT_SOCKET_PATH.into())
+    }
+}
+
+impl From<(&AppConfig, &Cli)> for Transport {
+    fn from((config, cli): (&AppConfig, &Cli)) -> Self {
+        let http_mode = config.grpc_http_mode.unwrap_or(cli.grpc_http_mode);
```

Comment:
Convention is usually that CLI should override config

Reply:
yes it is, but here `cli.grpc_http_mode` is a boolean flag and so it is always set (default to false if user dont set the flag)
So we cant check if the user explicilty set the flag to false or didnt set it at all, and so it can't take precedence over the config option

Reply:
TLDR:

`config.grpc_http_mode.unwrap_or(cli.grpc_http_mode)` is the same as

if cli.grpc_http_mode is true -> use it
if not fallback to the config value

Reply:
grpc_http_mode is Option<bool> right? Doesn't that mean that if the user has it set in the config then the cli will be ignored?

Reply:
No it is `bool` as any boolean flag when using clap, mean the flag is either set or not

Reply:
> Doesn't that mean that if the user has it set in the config then the cli will be ignored?

yes, but what's the deal? The default config is empty by default, use have to manually customize it

Reply:
> No it is bool as any boolean flag when using clap, mean the flag is either set or not

I mean the value from the config file is `Option<bool>`, leading to the surprising behaviour where the cli does nothing if it's set in the config file